crates/voicevox_core_c_api/src/lib.rs

```diff
@@ -692,6 +716,245 @@ pub extern "C" fn voicevox_error_result_to_message(
+#[no_mangle]
+pub unsafe extern "C" fn voicevox_user_dict_load(
+    user_dict: &VoicevoxUserDict,
+    dict_path: *const c_char,
+) -> VoicevoxResultCode {
+    into_result_code_with_error((|| {
+        let dict_path = ensure_utf8(unsafe { CStr::from_ptr(dict_path) })?;
+        let mut dict = user_dict.dict.lock().unwrap();
+        dict.load(dict_path)?;
+
+        Ok(())
+    })())
+}
+
+/// ユーザー辞書に単語を追加する
+/// @param [in] user_dict VoicevoxUserDictのポインタ
+/// @param [in] word 追加する単語
+/// @param [out] output_word_uuid 追加した単語のUUID
+/// @return 結果コード #VoicevoxResultCode
+///
+/// # Safety
+/// @param user_dict は有効な :VoicevoxUserDict のポインタであること
+///
+#[no_mangle]
+pub unsafe extern "C" fn voicevox_user_dict_add_word(
+    user_dict: &VoicevoxUserDict,
+    word: &VoicevoxUserDictWord,
+    output_word_uuid: NonNull<[u8; 16]>,
+) -> VoicevoxResultCode {
+    into_result_code_with_error((|| {
+        let word = word.try_into_word()?;
+        let uuid = {
+            let mut dict = user_dict.dict.lock().expect("lock failed");
+            dict.add_word(word)?
+        };
+        output_word_uuid.as_ptr().copy_from(uuid.as_bytes(), 16);
+
+        Ok(())
+    })())
+}
+
+/// ユーザー辞書の単語を更新する
+/// @param [in] user_dict VoicevoxUserDictのポインタ
+/// @param [in] word_uuid 更新する単語のUUID
+/// @param [in] word 新しい単語のデータ
+/// @return 結果コード #VoicevoxResultCode
+///
+/// # Safety
+/// @param user_dict は有効な :VoicevoxUserDict のポインタであること
+#[no_mangle]
+pub unsafe extern "C" fn voicevox_user_dict_update_word(
+    user_dict: &VoicevoxUserDict,
+    word_uuid: &[u8; 16],
+    word: &VoicevoxUserDictWord,
+) -> VoicevoxResultCode {
+    into_result_code_with_error((|| {
+        let word_uuid = Uuid::from_slice(word_uuid).map_err(CApiError::InvalidUuid)?;
+        let word = word.try_into_word()?;
+        {
+            let mut dict = user_dict.dict.lock().expect("lock failed");
+            dict.update_word(word_uuid, word)?;
+        };
+
+        Ok(())
+    })())
+}
+
+/// ユーザー辞書から単語を削除する
+/// @param [in] user_dict VoicevoxUserDictのポインタ
+/// @param [in] word_uuid 削除する単語のUUID
+/// @return 結果コード #VoicevoxResultCode
+#[no_mangle]
+pub extern "C" fn voicevox_user_dict_remove_word(
+    user_dict: &VoicevoxUserDict,
+    word_uuid: &[u8; 16],
+) -> VoicevoxResultCode {
+    into_result_code_with_error((|| {
+        let word_uuid = Uuid::from_slice(word_uuid).map_err(CApiError::InvalidUuid)?;
+        {
+            let mut dict = user_dict.dict.lock().expect("lock failed");
+            dict.remove_word(word_uuid)?;
+        };
+
+        Ok(())
+    })())
+}
+
+/// ユーザー辞書の単語をJSON形式で出力する
+/// @param [in] user_dict VoicevoxUserDictのポインタ
+/// @param [out] output_json JSON形式の文字列
+/// @return 結果コード #VoicevoxResultCode
+///
+/// # Safety
+/// @param user_dict は有効な :VoicevoxUserDict のポインタであること
+/// @param output_json 自動でheapメモリが割り当てられるので ::voicevox_json_free で解放する必要がある
+#[no_mangle]
+pub unsafe extern "C" fn voicevox_user_dict_get_json(
```

Comment:
project-vvm-async-apiの方針としては、"get"よりは"create"だと思います。この関数はオブジェクトをシリアライズしてJSONを作り出すものでもありますし。

```suggestion
pub unsafe extern "C" fn voicevox_user_dict_create_json(
```

Reply:
あるいは"to_"でもいいかもしれません。

```suggestion
pub unsafe extern "C" fn voicevox_user_dict_to_json(
```